docs/integrations/app-integrations/microsoft-teams.mdx

```diff
@@ -7,53 +7,162 @@ In this section, we present how to connect Microsoft Teams to MindsDB.
+If you installed MindsDB locally, make sure to install all Microsoft Teams dependencies by running `pip install .[teams]` or from the [requirements.txt](indsdb/integrations/handlers/ms_teams_handler/requirements.txt) file.
 </Tip>
 
 ## Usage
 
-Now, you can send messages to the configured Microsoft Teams channel:
+Now, you can post a message to a chat:
 
 ```sql
-INSERT INTO teams_datasource.messages(title, text)
-VALUES 
-('Hello from MindsDB!', 'This is a test message from MindsDB.');
+INSERT INTO teams_datasource.chat_messages (chatId, body_content)
+VALUES
+('your-chat-id', 'Hello from MindsDB!');
 ```
 
-You can also send multiple messages at once:
+You can also do the same for channels:
 
 ```sql
-INSERT INTO teams_datasource.messages(title, text)
-VALUES 
-('Hello from MindsDB!', 'This is a test message from MindsDB.'),
-('Hello again!', 'This is another test message from MindsDB.');
+SELECT * FROM teams_datasource.channels
 ```
 
-<Tip>
-you can use this handler to send outputs from models created in MinsDB as notifications to Microsoft Teams.
-</Tip>
\ No newline at end of file
+```sql
+INSERT INTO teams_datasource.channel_messages (channelIdentity_teamId, channelIdentity_channelId, body_content)
+VALUES
+('your-team-id', 'your-channel-id', 'Hello from MindsDB!');
+```
+
+## Supported Tables
+
+The following tables are supported by the Microsoft Teams handler:
+- `chats`: chats that the signed in user is a member of.
+- `chat_messages`: messages sent to the signed in user's chats.
+- `channels`: channels that the signed in user is a member of.
+- `channel_messages`: messages sent to the signed in user's channels.
+
+## Chatbot
+
+While the Microsoft Teams handler allows you to read/send messages to a chat or channel as shown above, it is also possible to create a chat bot that will listen to messages sent to a chat or channel and respond to them.
+
+### Step 1: Create a Microsoft Teams Data Source
+
+As shown above, create a database with the new `teams` engine by passing in the required parameters:
+
+```sql
+CREATE DATABASE teams_datasource
+WITH ENGINE = 'teams',
+PARAMETERS = {
+  "client_id": "your-client-id",
+  "client_secret": "your-client-secret",
+  "tenant_id": "your-tenant-id"
+};
+```
+
+<Note>
+The chat bot will assume the identity of the user who signed in with their Microsoft account when the database was created.
+</Note>
+
+### Step 2: Create an [Agent](https://docs.mindsdb.com/agents/agent)
+
+An agent is created by combining a [Model](https://docs.mindsdb.com/model-types) (conversational) with optional Skills.
+
+Here, we will create an agent without any skills, however, it is possilbe to create agents with skills such as the ability to answer questions from a knowledge base.
+
+#### Step 2.1: Create a Conversational Model
+
+Create a conversational model using either the `LangChain` or `LlamaIndex` integrations. Given below is an example of a conversational model created using the `LlamaIndex` integration:
```

Comment:
Let's link doc pages for LangChain and LlamaIndex here.

Reply:
@martyna-mindsdb Done.